### PR TITLE
COM-320: Refactor theming of `FilterBarButton`

### PR DESCRIPTION
**Pull request by juliawegmayr**
Screenshot 2024-01-17 at 14 12 42

### CODE REVIEW

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx [hunk 1]
```diff
@@ -1,58 +1,116 @@
...
 
     return (
-        <Button
-            className={clsx(className, classes.root, hasDirtyFields && classes.hasDirtyFields, openPopover && classes.open)}
-            disableRipple
-            endIcon={endIcon}
-            variant="outlined"
-            {...buttonProps}
-        >
+        <Root ownerState={ownerState} {...restProps} {...slotProps?.root} disableRipple endIcon={endIcon} variant="outlined">
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx:104]: Please add the `restProps` and `slotProps` to the end, so they can override the other values.

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx [hunk 2]
```diff
@@ -1,58 +1,123 @@
...
+})<{ ownerState: OwnerState }>(
+    ({ theme, ownerState }) => css`
+        position: relative;
+        cursor: pointer;
+        display: flex;
+        background-color: ${theme.palette.common.white};
+        border-color: ${theme.palette.grey[100]};
+        border-radius: 2px;
+
+        & .${buttonClasses.startIcon} .${svgIconClasses.root}, & .${buttonClasses.endIcon} .${svgIconClasses.root} {
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx:33]: I don't think the `&` is necessary here... or is it required for higher specificity? 🤔 

```suggestion
        .${buttonClasses.startIcon} .${svgIconClasses.root}, .${buttonClasses.endIcon} .${svgIconClasses.root} {
```

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx [hunk 3]
```diff
@@ -1,58 +1,123 @@
...
+        slotProps,
+        ...restProps
+    } = useThemeProps({
+        props: inProps,
+        name: "CometAdminFilterBarButton",
+    });
+
+    const ownerState: OwnerState = {
+        openPopover,
+        numberDirtyFields,
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx:100]: The number is not needed in the `ownerState`, it should be enough to pass in `hasDirtyFields`. 
This also makes sure the check in the styling (`${ownerState.numberDirtyFields &&`) stays the same.

**Review by jamesricky — CHANGES_REQUESTED** [review]

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx (hunk 2 above)
Comment (juliawegmayr) [reply packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx]: I corrected it to && as it is needed to overwrite the styling applied in MuiButton.ts

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx [hunk 4]
```diff
@@ -1,59 +1,121 @@
...
-    className,
-    ...buttonProps
-}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+export function FilterBarButtonWithStyles(inProps: FilterBarButtonProps) {
+    const {
+        children,
+        dirtyFieldsBadge,
+        numberDirtyFields,
+        openPopover,
+        classes,
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx:89]: `classes` should be removed with the new theming method. 

```suggestion
```

**Review by jamesricky — CHANGES_REQUESTED** [review]
Please make sure all deprecation notices stay directly above the deprecated things

File: packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx [hunk 5]
```diff
@@ -1,59 +1,119 @@
...
-    children,
-    dirtyFieldsBadge,
-    numberDirtyFields,
-    openPopover,
-    classes,
-    endIcon = <ChevronDown />,
-    className,
-    ...buttonProps
-}: FilterBarButtonProps & WithStyles<typeof styles>): React.ReactElement => {
+export function FilterBarButtonWithStyles(inProps: FilterBarButtonProps) {
```
Comment (jamesricky) [inline packages/admin/admin/src/table/filterbar/filterBarButton/FilterBarButton.tsx:83]: This should just be named `FilterBarButton` then the `export { FilterBarButtonWithStyles as FilterBarButton }` below can be removed.

**Review by jamesricky — CHANGES_REQUESTED** [review]
please fix the conflict and make sure the deprecation notices are back where they belong.

**Review by jamesricky — APPROVED** [review]

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**